scripts/india_plfs/wage_data/preprocess.py

```diff
@@ -0,0 +1,203 @@
+    },
+    {
+        "period": "2018-10",
+        "data_file": "Table_42_10_12_2018"
+    },
+    {
+        "period": "2019-01",
+        "data_file": "Table_42_01_03_2019"
+    },
+    {
+        "period": "2019-04",
+        "data_file": "Table_42_04_06_2019"
+    },
+]
+
+
+class PLFSWageDataLoader:
+    COLUMN_HEADERS = [
+        "period",
+        "territory",
+        "wage_rural_male",
+        "wage_rural_female",
+        "wage_rural_person",
+        "wage_urban_male",
+        "wage_urban_female",
+        "wage_urban_person",
+        "wage_total_male",
+        "wage_total_female",
+        "wage_total_person",
+    ]
+
+    def __init__(self, source, period):
+        self.source = source
+        self.period = period
+        self.raw_df = None
+        self.clean_df = None
+
+    def load(self):
+        df = pd.read_excel(self.source)
+        # Drop title rows in the top and  rows after 41.
+        # The actual data is between 4nd and 41st row. So keep only them.
+        df = df.iloc[4:41]
+        self.raw_df = df
+
+    def _setup_location(self):
+        self.clean_df["territory"] = self.clean_df["territory"].apply(
+            lambda x: INDIA_ISO_CODES[x])
+
+    def _make_column_numerical(self, column):
+        self.clean_df[column] = self.clean_df[column].astype(str).str.replace(
+            ",", "")
+        self.clean_df[column] = pd.to_numeric(self.clean_df[column])
+
+    def process(self):
+        # Set the date or period
+        self.clean_df = self.raw_df
+        self.clean_df.insert(loc=0, column="period", value=self.period)
+
+        # Rename columns
+        self.clean_df.columns = self.COLUMN_HEADERS
+
+        self._make_column_numerical("wage_rural_male")
+        self._make_column_numerical("wage_rural_female")
+        self._make_column_numerical("wage_rural_person")
+        self._make_column_numerical("wage_urban_male")
+        self._make_column_numerical("wage_urban_female")
+        self._make_column_numerical("wage_urban_person")
+        self._make_column_numerical("wage_total_male")
+        self._make_column_numerical("wage_total_female")
+        self._make_column_numerical("wage_total_person")
+
+        # Setup place ISO codes
+        self._setup_location()
+
+    def save(self, csv_file_path):
+        self.clean_df.to_csv(csv_file_path, index=False, header=True)
+
+
+def main():
+    """Runs the program."""
+    for dataset in DATASETS:
+        period = dataset["period"]
+        data_file = dataset["data_file"]
+        data_file_path = os.path.join(
+            os.path.dirname(__file__),
+            "data/{data_file}.xlsx".format(data_file=data_file),
+        )
+        loader = PLFSWageDataLoader(data_file_path, period)
+        csv_file_path = os.path.join(
+            os.path.dirname(__file__),
+            "./PLFSWageData_India_{data_file}.csv".format(data_file=data_file))
+
+        loader.load()
+        loader.process()
+        loader.save(csv_file_path)
```

Comment:
Thinking about importing the, could we change it to output a single CSV file?

The downstream pipeline configs will be a lot simpler.

Thanks!

Reply:
Done.

https://github.com/datacommonsorg/data/pull/390